src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/custom_vehicle_data_manager_impl.cc

```diff
@@ -0,0 +1,535 @@
+                       << policy_table::VehicleDataItem::kInteger
+                       << ", actual: " << item.getType());
+      return false;
+    }
+    auto value = item.asInt();
+    const auto minval =
+        GetOptional(item_schema.minvalue, std::numeric_limits<int64_t>::min());
+    const auto maxval =
+        GetOptional(item_schema.maxvalue, std::numeric_limits<int64_t>::max());
+    result = (minval <= value) && (maxval >= value);
+    if (!result) {
+      LOG4CXX_WARN(logger_,
+                   "Integer value: " << value
+                                     << " does not fit into boundaries : ("
+                                     << minval << " , " << maxval << ")");
+    }
+    return result;
+  }
+  if (item_type == policy_table::VehicleDataItem::kFloat) {
+    auto numeric_types = {smart_objects::SmartType::SmartType_Double,
+                          smart_objects::SmartType::SmartType_Integer,
+                          smart_objects::SmartType::SmartType_UInteger};
+    if (!helpers::in_range(numeric_types, item.getType())) {
+      LOG4CXX_WARN(logger_,
+                   "Parameter type mismatch. Expected: "
+                       << policy_table::VehicleDataItem::kFloat
+                       << ", actual: " << item.getType());
+      return false;
+    }
+    auto value = item.asDouble();
+    const auto minval =
+        GetOptional(item_schema.minvalue, std::numeric_limits<double>::min());
+    const auto maxval =
+        GetOptional(item_schema.maxvalue, std::numeric_limits<double>::max());
+    result = (minval <= value) && (maxval >= value);
+    if (!result) {
+      LOG4CXX_WARN(logger_,
+                   "Float value: " << value
+                                   << " does not fit into boundaries : ("
+                                   << minval << " , " << maxval << ")");
+    }
+    return result;
+  }
+  if (item_type == policy_table::VehicleDataItem::kBoolean) {
+    if (smart_objects::SmartType::SmartType_Boolean != item.getType()) {
+      LOG4CXX_WARN(logger_,
+                   "Parameter type mismatch. Expected: "
+                       << policy_table::VehicleDataItem::kBoolean
+                       << ", actual: " << item.getType());
+      return false;
+    }
+  }
+  return result;
+}
+
+bool CustomVehicleDataManagerImpl::ValidateStructTypeItem(
+    const ns_smart_device_link::ns_smart_objects::SmartObject& item,
+    const policy_table::VehicleDataItem& item_schema) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  auto find_param = [&item](const std::string& name) {
+    for (auto param : item.enumerate()) {
+      if (param == name) {
+        return item[param];
+      }
+    }
+    return smart_objects::SmartObject();
+  };
+
+  // Check for redundant parameters
+  for (const auto& param : item.enumerate()) {
+    auto param_schema = FindSchemaByNameRecursive(param);
+    if (!param_schema.is_initialized()) {
+      return false;
```

Comment:
In case if the mobile request contains parameter not specified neither in RPCspec nor in vehicle data items schema. The request will be rejected with INVALID_DATA.

Reply:
Before now, generally unknown parameters were just ignored, rather than rejected.

Reply:
Will be is not actual as soon as dynamic parameters will be added to validation schema.
Implementation will be ready for Friday 23 May

Reply:
Please see updates in commit 947c25585786f442a296fd4375c50d892684336b